File: app/src/main/java/de/karzek/diettracker/presentation/main/cookbook/recipeManipulation/adapter/viewHolder/RecipeManipulationIngredientsTitleAndPortionsViewHolder.java
package de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.viewHolder;

import android.support.v7.widget.RecyclerView;
import android.view.KeyEvent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.inputmethod.EditorInfo;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import de.karzek.diettracker.R;
import de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.itemWrapper.RecipeManipulationViewItemWrapper;
import de.karzek.diettracker.presentation.util.StringUtils;

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
public class RecipeManipulationIngredientsTitleAndPortionsViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.value_portions) EditText portions;

    private final OnPortionChangedListener onPortionChangedListener;

    public RecipeManipulationIngredientsTitleAndPortionsViewHolder(ViewGroup viewGroup,
                                                                   OnPortionChangedListener onPortionChangedListener) {
        super(LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.viewholder_recipe_man_ingredients_title_and_portions, viewGroup, false));
        ButterKnife.bind(this, itemView);

        this.onPortionChangedListener = onPortionChangedListener;

        setupPortionsChangedListener();
    }

    private void setupPortionsChangedListener() {
        portions.setOnEditorActionListener(new EditText.OnEditorActionListener() {
            @Override
            public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
                if (actionId == EditorInfo.IME_ACTION_DONE) {
                    onPortionChangedListener.onPortionChanges(Float.valueOf(portions.getText().toString()));
                }
                return true;
            }
        });
        portions.setOnFocusChangeListener(new View.OnFocusChangeListener() {
            @Override
            public void onFocusChange(View view, boolean b) {
                onPortionChangedListener.onPortionChanges(Float.valueOf(portions.getText().toString()));
            }
        });
    }

    public void bind(RecipeManipulationViewItemWrapper item) {
        portions.setText(StringUtils.formatFloat(item.getPortions()));
    }

    @OnClick(R.id.remove_portion) public void onRemovePortionClicked() {
        float newValue = Float.valueOf(portions.getText().toString()) -1;
        portions.setText(StringUtils.formatFloat(newValue));
        onPortionChangedListener.onPortionChanges(newValue);
    }

    @OnClick(R.id.add_portion) public void onAddPortionClicked() {
        float newValue = Float.valueOf(portions.getText().toString()) +1;
        portions.setText(StringUtils.formatFloat(newValue));
        onPortionChangedListener.onPortionChanges(newValue);
    }

    public interface OnPortionChangedListener {
        void onPortionChanges(float portion);
    }

}

File: app/src/main/java/de/karzek/diettracker/presentation/main/cookbook/recipeManipulation/adapter/viewHolder/RecipeManipulationIngredientsTitleAndPortionsViewHolder.kt
package de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.viewHolder

import android.view.View
import android.view.inputmethod.EditorInfo
import androidx.recyclerview.widget.RecyclerView
import de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.itemWrapper.RecipeManipulationViewItemWrapper
import de.karzek.diettracker.presentation.util.StringUtils
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.viewholder_recipe_man_ingredients_title_and_portions.*

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
class RecipeManipulationIngredientsTitleAndPortionsViewHolder(override val containerView: View,
                                                              private val onPortionChangedListener: OnPortionChangedListener) : RecyclerView.ViewHolder(containerView), LayoutContainer {

    private fun setupPortionsChangedListener() {
        value_portions.setOnEditorActionListener { _, actionId, _ ->
            if (actionId == EditorInfo.IME_ACTION_DONE) {
                onPortionChangedListener.onPortionChanges(java.lang.Float.valueOf(value_portions.text.toString()))
            }
            true
        }
        value_portions.onFocusChangeListener = View.OnFocusChangeListener { _, _ -> onPortionChangedListener.onPortionChanges(java.lang.Float.valueOf(value_portions.text.toString())) }
    }

    fun bind(item: RecipeManipulationViewItemWrapper) {
        value_portions.setText(StringUtils.formatFloat(item.portions!!))

        setupPortionsChangedListener()
        setupClickListeners()
    }

    private fun setupClickListeners() {
        remove_portion.setOnClickListener {
            val newValue = java.lang.Float.valueOf(value_portions.text.toString()) - 1
            value_portions.setText(StringUtils.formatFloat(newValue))
            onPortionChangedListener.onPortionChanges(newValue)
        }
        add_portion.setOnClickListener {
            val newValue = java.lang.Float.valueOf(value_portions.text.toString()) + 1
            value_portions.setText(StringUtils.formatFloat(newValue))
            onPortionChangedListener.onPortionChanges(newValue)
        }
    }

    interface OnPortionChangedListener {
        fun onPortionChanges(portion: Float)
    }

}
